changelog/31438.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+auth/cert: Do not poison the cache of trusted certificates if one or more certificates fails to load.
```

Comment:
```suggestion
auth/cert: Recover from partially populated caches of trusted certificates if one or more certificates fails to load.
```

We aren't injecting incorrect/bad entries within the cache which is what I'd associate cache poisoning